### PR TITLE
Fixes incorrect handling of cli and config file parameter merging

### PR DESCRIPTION
**Pull request by drewsonne**
Not sure when this was lost. This addresses #84 .

### CODE REVIEW

File: onelogin_aws_cli/argparse.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def __init__(self):
         self.add_argument(
             '-C', '--config-name',
             action=EnvDefault, required=False,
-            dest='config_name',
+            dest='config_name', default='defaults',
```
Comment (slycoder) [inline onelogin_aws_cli/argparse.py]: The stumbling block I ran into is that when you do it this way, and you try to retrieve the section, per configparser, the default section is not acknowledged by has_section so we'll try to add it.  However, adding a section with the name of the default section throws an exception.

Comment (drewsonne) [reply onelogin_aws_cli/argparse.py]: We can change it so it errors or returns `None` or something if there's no section. It doesn't need to be that smart.

If no section exists, we can make it explicit that a section would need to be added. Would that fix it?

Comment (slycoder) [reply onelogin_aws_cli/argparse.py]: That would but then it just moves a check somewhere else (see line 88 of my PR)

Comment (drewsonne) [reply onelogin_aws_cli/argparse.py]: Can you link to it? I can not see a line 88 in the diffs.

Comment (drewsonne) [reply onelogin_aws_cli/argparse.py]: Maybe moving the check is a good thing.

I propose we should only create a section when we run the `--configure` option. Any other time we should (gracefully) fail and say we don't have the section.

That is, only create a section when we explicitly want to create a new one. Any other time, say we don't have one.

Comment (drewsonne) [reply onelogin_aws_cli/argparse.py]: I think I've got this working now.

File: onelogin_aws_cli/argparse.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def __init__(self):
         )
 
         self.add_argument(
-            '-d', '--duration-seconds', type=int, default=3600,
+            '-d', '--duration-seconds', type=int,
```
Comment (slycoder) [inline onelogin_aws_cli/argparse.py]: What are the pros/cons of setting defaults via the config as you have versus just putting defaults here?

Comment (drewsonne) [reply onelogin_aws_cli/argparse.py]: If (as an example), a user were to have the `duration_seconds` value in the config file, and not provide it via the cli, as the cli has precedence to overwrite values in the config file, when we parse the args, this default value of `3600` will always take precedence over the values in the config file.

If the config precedence chain has the cli options higher than the config file, then defaults must be set at the bottom level of the chain (in this case, the config file).

File: onelogin_aws_cli/configuration.py [hunk 3]
```diff
@@ -7,24 +7,31 @@
...
+        super().__init__(
+            default_section='defaults',
+        )
 
         self.file = config_file
 
         if self.file is not None:
             self.load()
 
+    @property
```
Comment (slycoder) [inline onelogin_aws_cli/configuration.py]: All this jazz is no longer needed since we're never updating defaults with the fallbacks.

**Pull request merged**